From 2dded47b3202dfdf89aa96f84bf701b3d5acbe6c Mon Sep 17 00:00:00 2001
From: jim-p <jimp@netgate.com>
Date: Mon, 18 Feb 2019 10:25:24 -0500
Subject: [PATCH] Encoding for haproxy_listeners.php. Fixes #9335

Encode desc and table_actionsaclN before output.

(cherry picked from commit 3b40366aca55910b224ecf49d3fdacc9ad6c04f5)
---
 net/pfSense-pkg-haproxy-devel/Makefile                        | 2 +-
 .../files/usr/local/www/haproxy/haproxy_listeners.php         | 4 ++--
 net/pfSense-pkg-haproxy/Makefile                              | 2 +-
 .../files/usr/local/www/haproxy/haproxy_listeners.php         | 4 ++--
 4 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/net/pfSense-pkg-haproxy-devel/Makefile b/net/pfSense-pkg-haproxy-devel/Makefile
index 964d4a6f9e50..4653c0078d1e 100644
--- a/net/pfSense-pkg-haproxy-devel/Makefile
+++ b/net/pfSense-pkg-haproxy-devel/Makefile
@@ -2,7 +2,7 @@
 
 PORTNAME=	pfSense-pkg-haproxy-devel
 PORTVERSION=	0.59
-PORTREVISION=	16
+PORTREVISION=	17
 CATEGORIES=	net
 MASTER_SITES=	# empty
 DISTFILES=	# empty
diff --git a/net/pfSense-pkg-haproxy-devel/files/usr/local/www/haproxy/haproxy_listeners.php b/net/pfSense-pkg-haproxy-devel/files/usr/local/www/haproxy/haproxy_listeners.php
index 3639f13658ce..a8c75e62045f 100644
--- a/net/pfSense-pkg-haproxy-devel/files/usr/local/www/haproxy/haproxy_listeners.php
+++ b/net/pfSense-pkg-haproxy-devel/files/usr/local/www/haproxy/haproxy_listeners.php
@@ -367,7 +367,7 @@ function sort_sharedfrontends(&$a, &$b) {
 					<?=$frontend['name'];?>
 				  </td>
 				  <td>
-					<?=$frontend['desc'];?>
+					<?=htmlspecialchars($frontend['desc']);?>
 				  </td>
 				  <td>
 				    <?php
@@ -412,7 +412,7 @@ function sort_sharedfrontends(&$a, &$b) {
 							echo "<div title='{$hint}'>";
 							echo "<a href='haproxy_pool_edit.php?id={$backend}'>{$backend}</a>";
 							if (!empty($actionitem['acl'])) {
-								echo "&nbsp;if({$actionitem['acl']})";
+								echo "&nbsp;if(" . htmlspecialchars($actionitem['acl']) . ")";
 							}
 							echo "<br/></div>";
 						}
diff --git a/net/pfSense-pkg-haproxy/Makefile b/net/pfSense-pkg-haproxy/Makefile
index 8bfb32dd6e46..9310477b741d 100644
--- a/net/pfSense-pkg-haproxy/Makefile
+++ b/net/pfSense-pkg-haproxy/Makefile
@@ -2,7 +2,7 @@
 
 PORTNAME=	pfSense-pkg-haproxy
 PORTVERSION=	0.59
-PORTREVISION=	15
+PORTREVISION=	16
 CATEGORIES=	net
 MASTER_SITES=	# empty
 DISTFILES=	# empty
diff --git a/net/pfSense-pkg-haproxy/files/usr/local/www/haproxy/haproxy_listeners.php b/net/pfSense-pkg-haproxy/files/usr/local/www/haproxy/haproxy_listeners.php
index 3639f13658ce..a8c75e62045f 100644
--- a/net/pfSense-pkg-haproxy/files/usr/local/www/haproxy/haproxy_listeners.php
+++ b/net/pfSense-pkg-haproxy/files/usr/local/www/haproxy/haproxy_listeners.php
@@ -367,7 +367,7 @@ function sort_sharedfrontends(&$a, &$b) {
 					<?=$frontend['name'];?>
 				  </td>
 				  <td>
-					<?=$frontend['desc'];?>
+					<?=htmlspecialchars($frontend['desc']);?>
 				  </td>
 				  <td>
 				    <?php
@@ -412,7 +412,7 @@ function sort_sharedfrontends(&$a, &$b) {
 							echo "<div title='{$hint}'>";
 							echo "<a href='haproxy_pool_edit.php?id={$backend}'>{$backend}</a>";
 							if (!empty($actionitem['acl'])) {
-								echo "&nbsp;if({$actionitem['acl']})";
+								echo "&nbsp;if(" . htmlspecialchars($actionitem['acl']) . ")";
 							}
 							echo "<br/></div>";
 						}
